addons/account/models/account_payment.py

```diff
@@ -672,11 +672,16 @@ def _check_payment_method_line_id(self):
     # LOW-LEVEL METHODS
     # -------------------------------------------------------------------------
 
-    def new(self, values=None, origin=None, ref=None):
-        payment = super(AccountPayment, self.with_context(is_payment=True)).new(values, origin, ref)
-        if not payment.journal_id and not payment.default_get(['journal_id']):  # might not be computed because declared by inheritance
-            payment.move_id._compute_journal_id()
-        return payment
+    @api.onchange('available_journal_ids')
+    def _onchange_available_journal_ids(self):
+        """ This onchange is a temporary fix for two things:
+        1. When me remove the "new" inheritance method that to fix that changing the journl was not computng
+        dependant computed fields, now the "default journal" was not computed when creating new record. This onchange
+        fix it for now.
+        2. Fix the use case where a journal only suitable for one kind of operation (lets said inbound) is selected
+        and then the user selects "outbound" type, the journals remains selected."""
+        if not self.journal_id or self.journal_id not in self.available_journal_ids._origin:
```

Comment:
Because if not, m2m fields on onchanges and computed methods return a recordset of "new_ids" (they are actually existing ids, it's and ORM thing) but "self.journal_id" returns a recordset of ids. So the "not in" would always be False.
Another alternative would be:
```
        if not self.journal_id or self.journal_id.id not in self.available_journal_ids.ids:
            self.journal_id = self.available_journal_ids.ids[:1]
```

Do you prefer this alternative?

Reply:
Can't the `journal_id` be a `NewId` as well?